buildvertical.targets

```diff
@@ -8,102 +8,92 @@
     </MSBuild>
 
     <PropertyGroup>
-      <_SelectedOSGroup>%(VerticalTargets.OSGroup)</_SelectedOSGroup>
-      <_SelectedTargetGroup>%(VerticalTargets.TargetGroup)</_SelectedTargetGroup>
-      <_KeepProject>%(VerticalTargets.NoBuildConfigurations)</_KeepProject>
+      <_BestConfiguration>%(_Configuration.Identity)</_BestConfiguration>
+      <_KeepProject>false</_KeepProject>
+      <_KeepProject Condition="'%(_Configuration.KeepProject)' != ''">%(_Configuration.KeepProject)</_KeepProject>
     </PropertyGroup>
 
-    <Message Importance="Low" Text="Adding '$(_VerticalProject) KP:$(_KeepProject)', SelectedTargetGroup: $(_SelectedTargetGroup), SelectedOSGroup: $(_SelectedOSGroup)" />
-    <ItemGroup Condition="'$(_KeepProject)' != 'true'">
-      <Project Remove="$(_VerticalProject)" />
-      <Project Condition="'$(_SelectedTargetGroup)' != '' and '$(_SelectedOSGroup)' != ''" Include="$(_VerticalProjectName)">  
-        <TargetGroup>$(_SelectedTargetGroup)</TargetGroup>
-        <OSGroup>$(_SelectedOSGroup)</OSGroup>
-      </Project>
-      <Project Condition="'$(_SelectedTargetGroup)' == ''" Include="$(_VerticalProjectName)">
-        <OmitReason>No applicable TargetGroup</OmitReason>
-      </Project>
-      <Project Condition="'$(_SelectedOSGroup)' == ''" Include="$(_VerticalProjectName)">
-        <OmitReason>No applicable OSGroup</OmitReason>
-      </Project>
-      <Project Condition="'$(_OmitReason)' != ''">
-        <OmitReason>$(_OmitReason)</OmitReason>
-      </Project>
-    </ItemGroup>
-    <ItemGroup Condition="'$(_KeepProject)' == 'true'">
-      <Project Remove="$(_VerticalProject)" />
-      <Project Include="$(_VerticalProjectName)" />
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath) Adding $(_VerticalProject)  _Configuration: $(_BestConfiguration)" />
+    <Message Importance="Low" Text="NoBuildConfigurations: $(_KeepProject)" />
+    <PropertyGroup>
+       <OmitReason />
+       <OmitReason Condition="'$(_KeepProject)' != 'true' and '$(_BestConfiguration)' == ''">BuildConfigurations found but no applicable Configuration</OmitReason>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <OmitProject Include="$(MSBuildProjectFullPath)" Condition="'$(OmitReason)' != ''"/>
+      <_Configuration Condition="'%(_Configuration.Identity)' == 'None'" Remove="@(_Configuration)" />
     </ItemGroup>
+      
+    <Message Importance="Low" Condition="'$(OmitReason)' != ''" Text="$(MSBuildProjectFullPath) omitted because $(OmitReason)" />
   </Target>
 
   <!-- For the current project, target which determines compatibility for targets -->
   <Target Name="FindBestConfiguration"
-          Returns="@(VerticalTargets)"
+          Returns="@(_Configuration)"
           BeforeTargets="BeforeResolveReferences">
     
+    <Message Importance="Low" Text="FindBestConfiguration: $(MSBuildProjectFullPath) - $(BuildConfigurationImportFile)" />
     <MSBuild Targets="GetBuildConfigurations"
```

Comment:
Why do we need to call GetBuildConfigurations any longer here? We should just be able to get the BuildConfigurations property directly given we are in the project.